File: src/utils/api/config-manager.ts
/**
 * 配置管理模块统一 API 入口
 *
 * 后端 config_manager IPC 按 action 分发（get_config / apply_config）；
 * params 字段一律 camelCase。get_config_path / save_config_to_file 仍走独立 invoke。
 */

import { invoke } from '@tauri-apps/api/core'

/**
 * 调用 config_manager IPC
 * @param action 操作名称（取自 CONFIG_ACTIONS 常量）
 * @param params 参数对象（字段名使用 camelCase）
 */
export async function configManager<T = unknown>(
  action: string,
  params?: unknown,
): Promise<T> {
  return invoke<T>('config_manager', { req: { action, params: params ?? null } })
}

/**
 * 所有可用的 action 名称
 *
 * 与后端 `utils::config_manager::DISPATCHER` 注册的 action 一一对应。
 * 业务代码应优先使用此常量而非裸字符串，避免拼写错误。
 */
export const CONFIG_ACTIONS = {
  GET_CONFIG: 'get_config',
  APPLY_CONFIG: 'apply_config',
} as const

/** action 名称类型 */
export type ConfigAction = typeof CONFIG_ACTIONS[keyof typeof CONFIG_ACTIONS]

File: src-tauri/src/commands/system/config_manager.rs
//! 配置管理统一分发逻辑（system 域 config_manager 模块）
//!
//! 使用 `utils::dispatcher::Dispatcher` 注册式分发，2 个 action：
//! `get_config` / `apply_config`。两个命令均不需要 `AppHandle`。

use once_cell::sync::Lazy;
use serde::Deserialize;
use tauri::AppHandle;

use super::apply_config::{apply_config, get_config, ConfigEntry};
use crate::handler;
use crate::state::AppState;
use crate::utils::dispatcher::{ActionRequest, Dispatcher};

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
struct GetConfigParams {
    keys: Option<Vec<String>>,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
struct ApplyConfigParams {
    entries: Vec<ConfigEntry>,
}

static DISPATCHER: Lazy<Dispatcher> = Lazy::new(|| {
    let mut d = Dispatcher::new();

    d.register(
        "get_config",
        handler!(state, _app, params, {
            let p: GetConfigParams =
                serde_json::from_value(params).map_err(|e| format!("参数解析失败: {}", e))?;
            let r = get_config(&state, p.keys).await?;
            serde_json::to_value(r).map_err(|e| e.to_string())
        }),
    );

    d.register(
        "apply_config",
        handler!(state, _app, params, {
            let p: ApplyConfigParams =
                serde_json::from_value(params).map_err(|e| format!("参数解析失败: {}", e))?;
            apply_config(&state, p.entries).await?;
            Ok(serde_json::Value::Null)
        }),
    );

    d
});

/// 分发入口
pub async fn dispatch(
    state: AppState,
    app: AppHandle,
    req: ActionRequest,
) -> Result<serde_json::Value, String> {
    DISPATCHER.dispatch(state, app, req).await
}
